inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/permute_tile_8x8_4x4_fsv.cl

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "include/fetch.cl"
+#include "include/common.cl"
+#include "include/data_types.cl"
+
+#define unroll_for __attribute__((opencl_unroll_hint)) for
+#define CEIL_DIV(A, B) (((A) + (B) - 1) / (B))
+#define INPUT0_GET_TILED_INDEX(ORDER) INPUT0_GET_INDEX(ORDER)
+#define OUTPUT_GET_TILED_INDEX(ORDER) OUTPUT_GET_INDEX(ORDER)
+#define YZ_REMAINDER_LESS_THAN_TILE_SIZE ((YZ_REMAINDER_CONDITION) && (YZ_REMAINDER_SIZE < ( TILE_SIZE /2)))
+#define YZ_REMAINDER_MORE_THAN_TILE_SIZE ((YZ_REMAINDER_CONDITION) && (YZ_REMAINDER_SIZE >= ( TILE_SIZE /2)))
+
+#define INPUTVTYPE CAT(INPUT0_TYPE, TILE_SIZE)
+#define OUTPUTVTYPE CAT(OUTPUT_TYPE, TILE_SIZE)
+#define VLOAD CAT(vload, TILE_SIZE)
+#define VSTORE CAT(vstore, TILE_SIZE)
+#define AS_INPUTVTYPE CAT(as_, INPUTVTYPE)
+
+KERNEL (permute_tile_8x8_4x4_fsv)(
+    const __global INPUT0_TYPE* input,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+    )
+{
+#if INPUT0_DIMS == 4
+    const uint y = (get_global_id(1) / INPUT0_SIZE_X) * TILE_SIZE;
```

Comment:
It makes sense to cast get_global_id() result to int before any other operation, since get_global_id returns size_t type which is a bit complex to calculate

Reply:
Updated the kernel to cast all `get_global_id` and `get_local_id` calls to `uint`